deploy/AllocatorInit.sol

```diff
@@ -0,0 +1,278 @@
+        address gem0;
+        address gem1;
+        uint24  fee;
+        int24   tickLower;
+        int24   tickUpper;
+    }
+
+    function roles() external view returns (address);
+    function ilk() external view returns (bytes32);
+    function buffer() external view returns (address);
+    function deposit(LiquidityParams memory) external returns (uint128, uint256, uint256);
+    function withdraw(LiquidityParams memory, bool) external returns (uint256, uint256);
+    function collect(CollectParams memory) external returns (uint256, uint256);
+}
+
+interface StableSwapperLike {
+    function swapper() external view returns (address);
+}
+
+interface StableDepositorUniV3Like {
+    function depositor() external view returns (address);
+}
+
+interface ConduitMoverLike {
+    function ilk() external view returns (bytes32);
+    function buffer() external view returns (address);
+}
+
+interface KissLike {
+    function kiss(address) external;
+}
+
+struct AllocatorConfig {
+    uint256 debtCeiling;
+    address allocatorProxy;
+    uint8 facilitatorRole;
+    uint8 automationRole;
+    address facilitator;
+    address stableSwapperKeeper;
+    address stableDepositorUniV3Keeper;
+    address conduitMoverKeeper;
+    address[] swapTokens;
+    address[] depositTokens;
+    bytes32 vaultClKey;
+    bytes32 bufferClKey;
+    address ilkRegistry;
+    string ilkRegistryName;
+    string ilkRegistrySymbol;
+}
+
+library AllocatorInit {
+    uint256 constant WAD = 10 ** 18;
+    uint256 constant RAY = 10 ** 27;
+    uint256 constant RAD = 10 ** 45;
+
+    function switchOwner(address base, address currentOwner, address newOwner) internal {
+        if (currentOwner == newOwner) return;
+        require(WardsLike(base).wards(currentOwner) == 1, "AllocatorInit/current-owner-not-authed");
+        WardsLike(base).rely(newOwner);
+        WardsLike(base).deny(currentOwner);
+    }
+
+    function initShared(
+        DssInstance memory dss,
+        AllocatorSharedInstance memory sharedInstance
+    ) internal {
+        dss.chainlog.setAddress("ALLOCATOR_ORACLE",   sharedInstance.oracle);
+        dss.chainlog.setAddress("ALLOCATOR_ROLES",    sharedInstance.roles);
+        dss.chainlog.setAddress("ALLOCATOR_REGISTRY", sharedInstance.registry);
+    }
+
+    function initAllocator(
+        DssInstance memory dss,
+        AllocatorSharedInstance memory sharedInstance,
+        AllocatorNetworkInstance memory networkInstance,
+        AllocatorConfig memory cfg
+    ) internal {
+        bytes32 ilk = VaultLike(networkInstance.vault).ilk();
+
+        // Sanity checks
+        require(VaultLike(networkInstance.vault).roles()  == sharedInstance.roles,   "AllocatorInit/vault-roles-mismatch");
+        require(VaultLike(networkInstance.vault).buffer() == networkInstance.buffer, "AllocatorInit/vault-buffer-mismatch");
+        require(VaultLike(networkInstance.vault).vat()    == address(dss.vat),       "AllocatorInit/vault-vat-mismatch");
```

Comment:
In theory we would be missing to check the `nstJoin`, which should be previously in the chainlog and adapted in `dss-test` to come with the structure.
We can probably put it for now commented.

Reply:
done